### PR TITLE
fix: mutation tests (FPLA-373)

### PR DESCRIPTION
**Pull request by lewisbirks**
### JIRA link (if applicable) ###

[FPLA 373](https://tools.hmcts.net/jira/browse/FPLA-373)

### Change description ###

Re-enabled mutation tests on the nightly build.
Reduced current scope to just the mutation of code that can be covered in unit tests until the issue with jenkins is determined. The integration tests do work locally however.

Due to how some tests interact with the PIT testing environment they have been refactored to ensure that they can pass both normally and when the mutation test suite is running them.

#### How to run locally ####

The test suite can be run locally by when this branch is checked out by using the command `./gradlew pitest`

#### Current Coverage ####

Current mutation coverage without the integration tests:
- [81%](https://github.com/hmcts/fpl-ccd-configuration/files/4457160/unit-test-report.txt)

Current mutation coverage with integration tests:
- [89%](https://github.com/hmcts/fpl-ccd-configuration/files/4457161/unit%2Bintegration-test-report.txt)

Current mutation coverage threshold:
- 80% 

Note that the nightly-dev jenkins build will not trigger whilst this PR is open.

Last run can be found [here](https://build-beta.platform.hmcts.net/job/HMCTS_Nightly_FPL/job/fpl-ccd-configuration/job/nightly-dev/)

#### More Information ####

More information about PIT mutation testing can be found at the libraries website https://pitest.org/

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+@ActiveProfiles("integration-test")
+@WebMvcTest(DraftOrdersController.class)
+@OverrideAutoConfiguration(enabled = true)
+class DraftOrdersControllerAboutToStartTest extends AbstractControllerTest {
+    private static final Long CASE_ID = 1L;
+    private static final String SEND_DOCUMENT_EVENT = "internal-change:SEND_DOCUMENT";
+    private static final DocumentReference DOCUMENT_REFERENCE = DocumentReference.builder().build();
+
+    @Mock
+    ApplicationEventPublisher applicationEventPublisher;
```
Comment (tomaszpowroznik) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java:67]: private

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToStartEvent(caseDetails);
+
+        Stream.of(DirectionAssignee.values()).forEach(assignee ->
+            assertThat(callbackResponse.getData().get(assignee.toHearingDateField()))
+                .isEqualTo("1 January 2020, 12:00am"));
+    }
+
+    @Test
+    void shouldNotTriggerSDOEventWhenDraft() {
+        postSubmittedEvent(buildCallbackRequest(DRAFT));
```
Comment (tomaszpowroznik) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java:124]: class is about about-to-start event

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    @Test
+    void shouldNotTriggerSendDocumentEventWhenDraft() {
+        postSubmittedEvent(buildCallbackRequest(DRAFT));
+
+        verify(coreCaseDataService, never()).triggerEvent(any(), any(), any(), eq(SEND_DOCUMENT_EVENT), any());
+    }
+
+    @Test
+    void shouldTriggerSDOEventWhenSubmitted() throws Exception {
+        postSubmittedEvent(buildCallbackRequest(SEALED));
```
Comment (tomaszpowroznik) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java:138]: same

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java (hunk 2 above)
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java (hunk 3 above)
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java (hunk 1 above)
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToStartTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+                .collect(toList());
+
+        assertThat(localAuthorityDirections).isEqualTo(fullyPopulatedDirection);
+        AssertionsForClassTypes.assertThat(caseData.getStandardDirectionOrder().getOrderDoc()).isNotNull();
+        AssertionsForClassTypes.assertThat(caseData.getStandardDirectionOrder().getJudgeAndLegalAdvisor()).isNotNull();
+        AssertionsForClassTypes.assertThat(caseData.getJudgeAndLegalAdvisor()).isNull();
+    }
+
+    @Test
+    void aboutToSubmitShouldReturnErrorsWhenNoHearingDetailsExistsForSealedOrder() {
```
Comment (tomaszpowroznik) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java:138]: inconsistency, you renamed 
aboutToSubmitShouldPopulateHiddenCCDFieldsInStandardDirectionOrderToPersistData -> shouldPopulateHiddenCCDFieldsInStandardDirectionOrderToPersistData
which is fine, but not other method names

Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+        given(documentGeneratorService.generateDocmosisDocument(any(), any())).willReturn(docmosisDocument);
+    }
+
+    @Test
+    void shouldPopulateHiddenCCDFieldsInStandardDirectionOrderToPersistData() {
+        given(uploadDocumentService.uploadPDF(PDF, SEALED_ORDER_FILE_NAME))
+            .willReturn(document);
+
+        UUID uuid = UUID.randomUUID();
```
[the remark this answers is not in the document]
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+        assertThat(localAuthorityDirections).isEqualTo(fullyPopulatedDirection);
+        AssertionsForClassTypes.assertThat(caseData.getStandardDirectionOrder().getOrderDoc()).isNotNull();
+        AssertionsForClassTypes.assertThat(caseData.getStandardDirectionOrder().getJudgeAndLegalAdvisor()).isNotNull();
+        AssertionsForClassTypes.assertThat(caseData.getJudgeAndLegalAdvisor()).isNull();
+    }
+
+    @Test
+    void aboutToSubmitShouldReturnErrorsWhenNoHearingDetailsExistsForSealedOrder() {
+        given(uploadDocumentService.uploadPDF(PDF, SEALED_ORDER_FILE_NAME))
+            .willReturn(document());
```
[the remark this answers is not in the document]
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java]: done

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+@ActiveProfiles("integration-test")
+@WebMvcTest(DraftOrdersController.class)
+@OverrideAutoConfiguration(enabled = true)
+class DraftOrdersControllerAboutToSubmitTest extends AbstractControllerTest {
+    private static final Long CASE_ID = 1L;
+    private static final byte[] PDF = {1, 2, 3, 4, 5};
+    private static final DocumentReference DOCUMENT_REFERENCE = DocumentReference.builder().build();
+    private static final String SEALED_ORDER_FILE_NAME = "standard-directions-order.pdf";
+
+    Document document = document();
```
[the remark this answers is not in the document]
Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java]: done

**Pull request merged**